File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int low = 0;
        int high = nums.size() - 1;
        int minElem = 1001;

        while(low <= high){
            int mid = low + (high-low)/2;
            minElem = min(minElem, nums[mid]);

            if(nums[mid] >= nums[high]) low = mid + 1;
            else high = mid - 1;
        }

        return minElem;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        noel = {}
        for i in range(len(nums)):
            if nums[i] in noel:
                return True
            noel[nums[i]] = 1 + noel.get(nums[i], 0)
        return False